src/riscv/compiler.rs

```diff
@@ -197,10 +247,17 @@ instr branch_if_positive X, l: label {
     X + 2**32 - 1 = X_b1 + X_b2 * 0x100 + X_b3 * 0x10000 + X_b4 * 0x1000000 + wrap_bit * 2**32,
     pc' = wrap_bit * l + (1 - wrap_bit) * (pc + 1)
 }
+// input X is required to be the difference of two 32-bit unsigend values.
+// i.e. -2**32 < X < 2**32
+instr is_positive X -> Y {
+    X + 2**32 - 1 = X_b1 + X_b2 * 0x100 + X_b3 * 0x10000 + X_b4 * 0x1000000 + wrap_bit * 2**32,
+    Y = wrap_bit
+}
```

Comment:
If `X` is out of the pre-condition range, can't `wrap_bit` be non-binary?

Reply:
but wrap_bit is constrained to be binary somewhere else (unconditionally)